### PR TITLE
#OBA-264: Additional tests that validate partial inputs

### PR DESCRIPTION
**Pull request by elroman**

### CODE REVIEW

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java [hunk 1]
```diff
@@ -88,37 +88,104 @@ public SELF fintech_calls_list_accounts_for_anton_brueckner_no_ip_address() {
         return self();
     }
 
+    public SELF unknown_user_provided_initial_parameters_to_list_accounts_with_all_accounts_consent() {
+        String body = readResource("restrecord/tpp-ui-input/params/unknown-user-all-accounts-consent.json");
+
+        ExtractableResponse<Response> response = RestAssured
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java]: Please use the following formatting
```
given()
    .action()
when()
    .another_action()
```

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java [hunk 2]
```diff
@@ -181,6 +181,86 @@ void testAccountsListWithConsentUsingRedirectWithComputedIpAddress() {
...
+    void testTransactionsListWithConsentUsingEmbeddedForUnknownUser(Approach approach) {
+        given()
+                .embedded_mock_of_sandbox_for_max_musterman_transactions_running()
+                .preferred_sca_approach_selected_for_all_banks_in_opba(approach)
+                .rest_assured_points_to_opba_server();
+
+        when()
+                .fintech_calls_list_transactions_for_max_musterman()
+                .and()
+                .unknown_user_provided_initial_parameters_to_list_transactions_with_single_accounts_consent()
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java]: Maybe 
`user_max_musterman_provided_initial_parameters_to_list_transactions_but_without_psu_id_with_single_accounts_consent`

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java [hunk 3]
```diff
@@ -181,6 +181,86 @@ void testAccountsListWithConsentUsingRedirectWithComputedIpAddress() {
...
+                .and()
+                .user_max_musterman_provided_initial_psu_id_parameter_to_list_transactions_with_single_account_consent()
```
Comment (valb3r) [inline opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java]: Maybe 
`user_max_musterman_provided_psu_id_parameter_to_list_transactions_with_single_account_consent`
as it is not initial

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java (hunk 2 above)
Comment (elroman) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java [hunk 4]
```diff
@@ -88,37 +88,104 @@ public SELF fintech_calls_list_accounts_for_anton_brueckner_no_ip_address() {
...
-                    .header(X_REQUEST_ID, UUID.randomUUID().toString())
-                    .queryParam(REDIRECT_CODE_QUERY, redirectCode)
-                    .contentType(MediaType.APPLICATION_JSON_VALUE)
-                    .body(body)
-                .when()
-                    .post(AUTHORIZE_CONSENT_ENDPOINT, serviceSessionId)
-                .then()
-                    .statusCode(HttpStatus.ACCEPTED.value())
-                .extract();
+                                                         .given()
```
[the remark this answers is not in the document]
Comment (elroman) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java [hunk 5]
```diff
@@ -88,37 +88,104 @@ public SELF fintech_calls_list_accounts_for_anton_brueckner_no_ip_address() {
...
         updateRedirectCode(response);
 
         return self();
     }
+
+    public SELF user_anton_brueckner_provided_initial_parameters_to_list_accounts_with_wrong_ibans() {
+        String body = readResource("restrecord/tpp-ui-input/params/anton-brueckner-account-wrong-ibans.json");
+
+        ExtractableResponse<Response> response = RestAssured
+                                                         .given()
```
[the remark this answers is not in the document]
Comment (elroman) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java (hunk 1 above)
Comment (elroman) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/main/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/mocks/WiremockAccountInformationRequest.java]: done

File: opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java (hunk 3 above)
Comment (elroman) [reply opba-protocols/xs2a-protocol-tests/xs2a-bdd-wiremock/src/test/java/de/adorsys/opba/protocol/xs2a/tests/e2e/wiremock/WiremockE2EXs2aProtocolTest.java]: done

**Pull request merged**